Mathlib/Algebra/Group/Pi/Sum.lean

```diff
@@ -0,0 +1,110 @@
+/-
+Copyright (c) 2024 Amelia Livingston. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Amelia Livingston
+-/
+
+import Mathlib.Algebra.Group.Defs
+
+/-!
+# Monoid, group etc structures on families indexed by sum types
+
+Given 2 families of e.g. groups `αᵢ, βᵢ` indexed by `ια, ιβ` respectively, the components
+of the resulting family of types indexed by `ια ⊕ ιβ` each have a group structure.
+
+# Implementation notes
+
+A little care is required to avoid non-defeq diamonds - e.g. `instElimMulOneClass` can't just be
+defined with a match expression, since the resulting `One` and `Mul` instances won't be defeq to
+`instElimOne` and `instElimMul`.
+
+# Todo
+
+Missing some of the algebraic hierarchy compared to e.g. `Mathlib.Algebra.Group.Prod`.
```

Comment:
```suggestion
## Implementation notes

A little care is required to avoid non-defeq diamonds - e.g. `instElimMulOneClass` can't just be
defined with a match expression, since the resulting `One` and `Mul` instances won't be defeq to
`instElimOne` and `instElimMul`.

## Todo

Missing some of the algebraic hierarchy compared to e.g. `Mathlib.Algebra.Group.Prod`.
```